### DISCUSSION TITLE
How to structure the data when I need filtering by user id?

### DISCUSSION THREAD

**Discussion by sdaza** [microsoft/AI-Chat-App-Hack#42] [Q&A]
Hello, I am working on a project where I need to retrieve information from users' notes, such as topics of discussion. I have a database that contains user ID numbers, dates of interaction, and notes. Each row represents an interaction, and there can be multiple interactions per user.

For example, if I have a prompt that says `summarize notes from user 12345` I want to ensure that the prompt only selects notes from user 12345 with a probability of 1.

I'm not sure how to structure this project exactly. Here's what I'm thinking:

1. First, I need to create a method to convert my tabular data and text (notes, just a column in my table) into vectors.
2. Then, I believe I would need to combine the vectorized data with keyword searches (?), or filtering, I don't know.

Do you have any ideas or examples that I can refer to? I would greatly appreciate any help. Thank you!

**Comment by pamelafox** [comment]
Hi @sdaza!

I would suggest utilizing GPT function calling (now known as "tools") to perform those more specific queries. Here's an example where someone extended the chat approach to optionally search GitHub issues:
https://gist.github.com/pamelafox/a3fdea186b687509c02cb186ca203328#file-chatreadretrieveread-py-L108

The idea is that you first ask the LLM to pick what function it will use, the generic search_data function or the more specific search_for_user function. The LLM will hopefully select search_for_user, so when it returns search_for_user(12345), you'd then use that to perform a search with a filter.

Anthony will be talking about that approach at a session next week as well.

Let me know if that makes sense. Here's more on function calling: https://platform.openai.com/docs/guides/function-calling

**Comment by sdaza** [comment]
Thanks, @pamelafox! This is helpful. I need to study this further and first figure out how to import the data into Azure :)

**Reply by sdaza** [reply]
Hi @pamelafox, for my project, I  would like to extract specific information from doctor's notes. Again, these notes are nested  by user (id). Let's say I want to get, based on the notes, if the user will have a next session. 

```
user date session doc note
1       1      1     Text
1       2      2     Text
1       3      3     Text
```

A good idea will be trying to retrieve information from last sessions, and try to figure out what the doctor says, filtering first by the user id. Do you have any ideas on how to approach this particular use case?

Thanks so much! I am very new to this :)
